### DISCUSSION TITLE
\* Pocketbase Update */

### DISCUSSION THREAD

**Discussion by Ianmello10** [pocketbase/pocketbase#1126] [Q&A] [answered]
suppose I have my pocketbase in production and tomorrow a new version comes out, how would I be able to update the pocketbase without breaking it in production and losing my data?

**Comment by ganigeorgiev** [comment, marked as answer]
**Please note that PocketBase is still under active development and backward compatibility is not guaranteed before reaching v1 and therefore is not suitable at the moment for critical production applications, unless you are OK with putting an extra effort to perform manual migration steps from time to time.**

With that said, before each PocketBase deployment I recommended to always have a backup of your production `pb_data` (eg. `cp -r ./pb_data ./pb_data_backup`) in case something goes wrong.

With each release, I try to list the changes in the [GitHub release notes](https://github.com/pocketbase/pocketbase/releases) and/or in the [CHANGELOG.md](https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md).

During development, if you use PocketBase as framework, you can write migration scripts with Go and embed them in your final executable - https://pocketbase.io/docs/migrations/. Migrations are automatically applied on server start.
I'm currently also experimenting with different approaches to allow similar experience when the prebuilt PocketBase executable is used and most likely with the next v0.9 release you'll be able to write migration scripts with JavaScript.


---

So to answer you question - when new PocketBase version is released, currently you'll have to:

- read the changelog and apply any necessary changes in your code
- upload the new binary (_at least on linux, you don't need to first stop the server when removing the old binary, because the os will keep the old inode until there is a process using it, so it is safe to call `rm`, `mv`, `unlink`, etc. before stopping the server. `rsync` is also safe, unless you use the `rsync --in-place` flag_)
- backup your `pb_data` (_for transactional safety you could kill the running process before the copy_)
- restart the server process
